flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -556,7 +556,14 @@ public void setExecutionMode(ExecutionMode executionMode) {
      * <p>The default execution mode is {@link ExecutionMode#PIPELINED}.
      *
      * @return The execution mode for the program.
+     * @deprecated All Flink DataSet APIs are deprecated since Flink 1.18 and will be removed in a
+     *     future Flink major version. You can still build your application in DataSet, but you
+     *     should move to either the DataStream and/or Table API.
+     * @see <a href="https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=158866741">
+     *     FLIP-131: Consolidate the user-facing Dataflow SDKs/APIs (and deprecate the DataSet
+     *     API</a>
      */
+    @Deprecated
```

Comment:
Shouldn't we also deprecated the class `ExecutionMode` and the method `setExecutionMode`?

Reply:
Also, there are some descriptions about the execution mode in the class JavaDoc, which should also be updated.

Reply:
1. `ExecutionMode` and `setExecutionMode ` are both deprecated.
2. The class JavaDoc of execution mode is updated.